src/test/regress/expected/merge_unsupported.out

```diff
@@ -0,0 +1,106 @@
+DROP TABLE IF EXISTS source;
+NOTICE:  table "source" does not exist, skipping
+CREATE TABLE target (tid integer, balance integer)
+  WITH (autovacuum_enabled=off);
+CREATE TABLE source (sid integer, delta integer) -- no index
+  WITH (autovacuum_enabled=off);
+SELECT citus_add_local_table_to_metadata('target');
+ citus_add_local_table_to_metadata
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT citus_add_local_table_to_metadata('source');
+ citus_add_local_table_to_metadata
+---------------------------------------------------------------------
+
+(1 row)
+
+\set SHOW_CONTEXT errors
+-- used in a CTE
+WITH foo AS (
+  MERGE INTO target USING source ON (true)
+  WHEN MATCHED THEN DELETE
+) SELECT * FROM foo;
+ERROR:  WITH query "foo" does not have a RETURNING clause
+-- used in COPY
+COPY (
+  MERGE INTO target USING source ON (true)
+  WHEN MATCHED THEN DELETE
+) TO stdout;
+ERROR:  MERGE not supported in COPY
+-- used in a CTE with RETURNING
+WITH foo AS (
+  MERGE INTO target USING source ON (true)
+  WHEN MATCHED THEN DELETE RETURNING target.*
+) SELECT * FROM foo;
+ERROR:  MERGE with RETURNING is not yet supported
+-- used in COPY with RETURNING
+COPY (
+  MERGE INTO target USING source ON (true)
+  WHEN MATCHED THEN DELETE RETURNING target.*
+) TO stdout;
+ERROR:  MERGE not supported in COPY
+-- unsupported relation types
+-- view
+CREATE VIEW tv AS SELECT count(tid) AS tid FROM target;
+MERGE INTO tv t
+USING source s
+ON t.tid = s.sid
+WHEN NOT MATCHED THEN
+	INSERT DEFAULT VALUES;
+ERROR:  cannot insert into view "tv"
+DETAIL:  Views that return aggregate functions are not automatically updatable.
+HINT:  To enable inserting into the view using MERGE, provide an INSTEAD OF INSERT trigger.
+DROP VIEW tv;
+CREATE TABLE sq_target (tid integer NOT NULL, balance integer)
+  WITH (autovacuum_enabled=off);
+CREATE TABLE sq_source (delta integer, sid integer, balance integer DEFAULT 0)
+  WITH (autovacuum_enabled=off);
+SELECT citus_add_local_table_to_metadata('sq_target');
+ citus_add_local_table_to_metadata
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT citus_add_local_table_to_metadata('sq_source');
+ citus_add_local_table_to_metadata
+---------------------------------------------------------------------
+
+(1 row)
+
+INSERT INTO sq_target(tid, balance) VALUES (1,100), (2,200), (3,300);
+INSERT INTO sq_source(sid, delta) VALUES (1,10), (2,20), (4,40);
+CREATE VIEW v AS SELECT * FROM sq_source WHERE sid < 2;
+-- RETURNING
+BEGIN;
+INSERT INTO sq_source (sid, balance, delta) VALUES (-1, -1, -10);
+MERGE INTO sq_target t
+USING v
+ON tid = sid
+WHEN MATCHED AND tid > 2 THEN
+    UPDATE SET balance = t.balance + delta
+WHEN NOT MATCHED THEN
+        INSERT (balance, tid) VALUES (balance + delta, sid)
+WHEN MATCHED AND tid < 2 THEN
+        DELETE
+RETURNING *;
+ERROR:  MERGE with RETURNING is not yet supported
+ROLLBACK;
```

Comment:
Can we move the failing query in pg15 into merge_unsupported.sql ?
https://github.com/citusdata/citus/actions/runs/12158205599/attempts/1#summary-33905775016
It looks like the failing pg15 queries are covered by scenarios in merge_unsupported (CTE, Copy) so perhaps they could be removed from pg15 ?

Reply:
Yes, removed it. BTW we shouldn't be getting this below diff with my changes. Is the above run with my changes?

<img width="337" alt="image" src="https://github.com/user-attachments/assets/513d7a14-5964-43cb-9738-63858450c7b9" />

Reply:
I think I was seeing a diff in pg15 when running your dev branch in my local environment, but I see that pg15 is passing with your latest changes

Reply:
https://github.com/citusdata/citus/actions/runs/12310062179/job/34358136187?pr=7781 - pg15 is ok